sam_view.c

```diff
@@ -290,7 +292,18 @@ int main_samview(int argc, char *argv[])
                 srand(settings.subsam_seed);
                 settings.subsam_seed = rand();
             }
-            settings.subsam_frac = strtod(q, &q);
+ 
+            if (q && *q == '.') {
+                settings.subsam_frac = strtod(q, &q);
+                if (*q) ret = 1;
+            } else {
+                ret = 1;
+            }
+
+            if (ret == 1) {
+                print_error_errno("view", "Incorrect sampling argument \"%s\"", optarg);
```

Comment:
errno will almost never mean anything here, so just use `print_error`. Also the `fprintf` added below would be better as `print_error` — prettifying these error messages is what the `print_error` family is for.